python/tidy/tidy.py

```diff
@@ -376,6 +381,38 @@ def check_ruff_lints():
             )
 
 
+@dataclass
+class PyreflyDiagnostic:
+    """
+    Represents a single diagnostic error reported by Pyrefly.
+    """
+
+    line: int
+    column: int
+    stop_line: int
+    stop_column: int
+    path: str
+    code: int
+    name: str
+    description: str
+    concise_description: str
+
+
+def run_python_type_checker() -> Iterator[Tuple[str, int, str]]:
+    print("\r ➤  Running `pyrefly` checks...")
+    try:
+        result = subprocess.run(["pyrefly", "check", "--output-format", "json"], capture_output=True)
+        parsed_json = json.loads(result.stdout)
+        errors = parsed_json.get("errors", [])
+    except subprocess.CalledProcessError as error:
+        print(f"{colorama.Fore.YELLOW}{error}{colorama.Style.RESET_ALL}")
+        pass
+    else:
+        for error in errors:
+            diagnostic = PyreflyDiagnostic(**error)
+            yield normalize_path(diagnostic.path), diagnostic.line, diagnostic.name
```

Comment:
`diagnostic.name` here just show the code of the error e.g `bad-override` when typecheck fails at a given line, but it would be useful to show more information l(ike we do for the ruff lint) so the user has more context e.g using the information in `diagnostic.concise_description` or `diagnostic.description` as well.